### ISSUE TITLE
Chained call expressions are not indented.

### ISSUE THREAD

**Issue by m59peacemaker** [eslint/eslint#11320]
**What rule do you want to change?**
`indent`

**Does this change cause the rule to produce more or fewer warnings?**
Fewer.

**How will the change be implemented? (New option, new default behavior, etc.)?**
New default behavior with new option to configure it.

**Examples:**

**incorrect** code
```js
foo
(123)
(456)
```
**correct** code
```js
foo
  (123)
  (456)
```

**What does the rule currently do for this code?**
Chaining function invocations across multiple lines (when permitted by func-call-spacing) expects no indentation.

**What will the rule do after it's changed?**
Chaining function invocations should be indented by default in the same manner as chaining member method invocations, with the same options for configuration.

```js
foo
  .bar(123)
  .baz(456)

foo
  (123)
  (456)
```

**Are you willing to submit a pull request to implement this change?**
I am willing to try!

**Comment by kaicataldo** [comment]
This seems like a reasonable enhancement to me - the "correct" example above is what I would expect the rule to enforce by default. Could the option be part of the `CallExpression` config object?

**Comment by m59peacemaker** [comment]
Other than whether it should be `parenthesis`, `parentheses` or just `parens`, that sounds like the perfect place for it.

**Comment by kaicataldo** [comment]
Exploring this a bit more, `no-multiline-expression` actually warns on this case (demo example [here](https://eslint.org/demo/#eyJ0ZXh0IjoiZm9vXHJcbiAoYmFyKVxyXG4gIiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo5LCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7ImNvbnN0cnVjdG9yLXN1cGVyIjoyLCJmb3ItZGlyZWN0aW9uIjoyLCJnZXR0ZXItcmV0dXJuIjoyLCJuby1jYXNlLWRlY2xhcmF0aW9ucyI6Miwibm8tY2xhc3MtYXNzaWduIjoyLCJuby1jb21wYXJlLW5lZy16ZXJvIjoyLCJuby1jb25kLWFzc2lnbiI6Miwibm8tY29uc29sZSI6Miwibm8tY29uc3QtYXNzaWduIjoyLCJuby1jb25zdGFudC1jb25kaXRpb24iOjIsIm5vLWNvbnRyb2wtcmVnZXgiOjIsIm5vLWRlYnVnZ2VyIjoyLCJuby1kZWxldGUtdmFyIjoyLCJuby1kdXBlLWFyZ3MiOjIsIm5vLWR1cGUtY2xhc3MtbWVtYmVycyI6Miwibm8tZHVwZS1rZXlzIjoyLCJuby1kdXBsaWNhdGUtY2FzZSI6Miwibm8tZW1wdHktY2hhcmFjdGVyLWNsYXNzIjoyLCJuby1lbXB0eS1wYXR0ZXJuIjoyLCJuby1lbXB0eSI6Miwibm8tZXgtYXNzaWduIjoyLCJuby1leHRyYS1ib29sZWFuLWNhc3QiOjIsIm5vLWV4dHJhLXNlbWkiOjIsIm5vLWZhbGx0aHJvdWdoIjoyLCJuby1mdW5jLWFzc2lnbiI6Miwibm8tZ2xvYmFsLWFzc2lnbiI6Miwibm8taW5uZXItZGVjbGFyYXRpb25zIjoyLCJuby1pbnZhbGlkLXJlZ2V4cCI6Miwibm8taXJyZWd1bGFyLXdoaXRlc3BhY2UiOjIsIm5vLW1peGVkLXNwYWNlcy1hbmQtdGFicyI6Miwibm8tbmV3LXN5bWJvbCI6Miwibm8tb2JqLWNhbGxzIjoyLCJuby1vY3RhbCI6Miwibm8tcmVkZWNsYXJlIjoyLCJuby1yZWdleC1zcGFjZXMiOjIsIm5vLXNlbGYtYXNzaWduIjoyLCJuby1zcGFyc2UtYXJyYXlzIjoyLCJuby10aGlzLWJlZm9yZS1zdXBlciI6Miwibm8tdW5leHBlY3RlZC1tdWx0aWxpbmUiOjIsIm5vLXVucmVhY2hhYmxlIjoyLCJuby11bnNhZmUtZmluYWxseSI6Miwibm8tdW5zYWZlLW5lZ2F0aW9uIjoyLCJuby11bnVzZWQtbGFiZWxzIjoyLCJuby11c2VsZXNzLWVzY2FwZSI6MiwicmVxdWlyZS15aWVsZCI6MiwidXNlLWlzbmFuIjoyLCJ2YWxpZC10eXBlb2YiOjIsInJlcXVpcmUtanNkb2MiOjIsInZhbGlkLWpzZG9jIjoyLCJuby1tYWdpYy1udW1iZXJzIjoyfSwiZW52Ijp7fX19)).

If ASI would add a semicolon here, the current behavior is actually correct.

**Comment by m59peacemaker** [comment]
It's a critical issue for me, as my primary paradigm is dominated with curried functions that have chained invocations in this way. Single line expressions in this manner are often far worse than the lack of indentation or having to keep the indent rule disabled. [Here is a thread about a code style for which multiline call expression indentation is necessary.](https://github.com/sanctuary-js/sanctuary/issues/438).
